### PR TITLE
Adds few more gitlab group options

### PR DESCRIPTION
**Pull request by mkschuel**
##### SUMMARY
Enhanced existing community.general.gitlab_group to handle additional gitlab properties:
- project_creation_level
- auto_devops_enabled
- subgroup_creation_level

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
gitlab_group

##### ADDITIONAL INFORMATION
Just adding few more options to configure a GitLab group properly. Has been tested on a local instance.

### CODE REVIEW

File: plugins/modules/source_control/gitlab/gitlab_group.py [hunk 1]
```diff
@@ -61,6 +61,22 @@
     choices: ["private", "internal", "public"]
     default: private
     type: str
+  project_creation_level:
+    description:
+      - Determine if developers can create projects in the group
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_group.py:66]:
```suggestion
      - Determine if developers can create projects in the group.
```
(same for the others)

File: plugins/modules/source_control/gitlab/gitlab_group.py [hunk 2]
```diff
@@ -61,6 +61,22 @@
...
+    choices: ["developer", "maintainer", "noone"]
+    default: developer
+    type: str
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_group.py:69]:
```suggestion
    type: str
    version_added: 3.5.0
```
(same for the others)

File: plugins/modules/source_control/gitlab/gitlab_group.py [hunk 3]
```diff
@@ -166,7 +182,10 @@ def createOrUpdateGroup(self, name, parent, options):
                 'name': name,
                 'path': options['path'],
                 'parent_id': parent_id,
-                'visibility': options['visibility']
+                'visibility': options['visibility'],
+                'project_creation_level': options['project_creation_level'],
+                'auto_devops_enabled': options['auto_devops_enabled'],
+                'subgroup_creation_level': options['subgroup_creation_level']
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_group.py:188]:
```suggestion
                'subgroup_creation_level': options['subgroup_creation_level'],
```

File: plugins/modules/source_control/gitlab/gitlab_group.py [hunk 4]
```diff
@@ -258,6 +280,9 @@ def main():
         state=dict(type='str', default="present", choices=["absent", "present"]),
         parent=dict(type='str'),
         visibility=dict(type='str', default="private", choices=["internal", "private", "public"]),
+        project_creation_level=dict(type='str', default="developer", choices=["developer", "maintainer", "noone"]),
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_group.py:283]: Please use `'` for all strings (and not `"` for some). This is already inconsequent for existing options, there's no need to keep doing that for new ones :)

**Comment by mkschuel** [comment]
@felixfontein: thank you for your review. Your suggestions were implemented.

**Comment by felixfontein** [comment]
Please don't forget the "(same for the others)" parts :)

**Comment by mkschuel** [comment]
> Please don't forget the "(same for the others)" parts :)

Sorry, but do I get it right? I should change the `"` also in the line which were already there?

**Comment by felixfontein** [comment]
Can you take a look at the unit tests (tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py) and see how hard it would be to add tests for this? Also having integration tests for this (tests/integration/targets/gitlab_group) would be nice. (The unit tests are executed in CI, the integration tests are not.)

**Comment by mkschuel** [comment]
> Can you take a look at the unit tests (tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py) and see how hard it would be to add tests for this? Also having integration tests for this (tests/integration/targets/gitlab_group) would be nice. (The unit tests are executed in CI, the integration tests are not.)

I'll be on vacation for the next two weeks. Would be great to get this merged to have it available for our further development and I'll do the test stuff as follow-up after my vacation.

**Comment by felixfontein** [comment]
I think it's better to have tests included in the feature PR.

**Comment by zanssa** [comment]
Thanks for the contribution. :) 

Can you also add or modify one example to include these new parameters?

**Comment by mkschuel** [comment]
Merge on merge killed me.

**Pull request closed without merging**